crates/moonrun/src/main.rs

```
@@ -193,6 +193,39 @@ fn read_char(
     }
 }
 
+fn read_bytes_from_stdin(
+    scope: &mut v8::HandleScope,
+    _args: v8::FunctionCallbackArguments,
+    mut ret: v8::ReturnValue,
+) {
+    let mut buffer = Vec::new();
+    let stdin = io::stdin();
+    let mut handle = stdin.lock();
+
+    match handle.read_to_end(&mut buffer) {
+        Ok(size) => {
+            if size == 0 {
+                let empty_array_buffer = v8::ArrayBuffer::new(scope, 0);
+                let empty_uint8_array =
+                    v8::Uint8Array::new(scope, empty_array_buffer, 0, 0).unwrap();
+                ret.set(empty_uint8_array.into());
+            } else {
+                let array_buffer = v8::ArrayBuffer::new(scope, size);
+                let uint8_array = v8::Uint8Array::new(scope, array_buffer, 0, size).unwrap();
+                unsafe {
+                    std::ptr::copy(buffer.as_ptr(), get_array_buffer_ptr(array_buffer), size);
+                }
+                ret.set(uint8_array.into());
+            }
+        }
+        Err(_) => {
+            let empty_array_buffer = v8::ArrayBuffer::new(scope, 0);
```

maybe we should report the error to user?